src/main/java/org/tools4j/hoverraft/state/LeaderState.java

```diff
@@ -99,13 +104,70 @@ private Transition onTimerEvent(final ServerContext serverContext, final TimerEv
         return Transition.STEADY;
     }
 
-    private void updateCommitIndex(final ServerContext serverContext) {
-        //FIXME impl
+    //FIXme test it!!!
+    private void updateCommitIndex() {
+
+        long currentCommitIndex = volatileState().commitIndex();
+
+        int followerCount = volatileState().followerCount();
+
+        final int majority = (followerCount + 1) / 2;
+
+        long minIndexHigherThanCommitIndex = Long.MAX_VALUE;
+        int higherThanCommitCount = 0;
+
+        for(int idx = 0; idx < followerCount; idx++) {
+
+            final TrackedFollowerState followerState = volatileState().followerState(idx);
+            final long matchIndex = followerState.matchIndex();
+
+            if (matchIndex > currentCommitIndex) {
+
+                persistentState().commandLog().readIndex(matchIndex);
+                final int matchTerm = persistentState().commandLog().readTerm();
```

Comment:
the matchTerm is for current match index which is for currently iterated TrackingFollowerState. I think this logic can be abstracted.